modules/core/src/transitions/gpu-spring-transition.js

```diff
@@ -0,0 +1,240 @@
+    // line up the values we're transitioning from with the new values we're
+    // transitioning to.
+    const precisionMultiplier = this.attribute.doublePrecision ? 2 : 1;
+    const toLength =
+      (this.attribute.userData.noAlloc
+        ? this.attribute.value.length
+        : numInstances * this.attribute.size) * precisionMultiplier;
+    if (this.buffers.current) {
+      this.buffers = this._updateExistingBuffers(
+        gl,
+        this.buffers,
+        this.attribute,
+        toLength,
+        transitionSettings.enter
+      );
+
+      // when an attribute changes values, a new transition is started. These
+      // are properties that we have to store on this instance but can change
+      // when new transitions are started, so we have to keep them up-to-date. :(
+      this.transitionSettings = transitionSettings;
+      this.currentBufferLayout = this.attribute.bufferLayout;
+    } else {
+      const data = this.attribute.constant
+        ? this.attribute.getValue()
+        : this.attribute.getBuffer().getData({});
+      const usage = GL.DYNAMIC_COPY;
+      this.buffers.previous = new Buffer(gl, {data, usage});
+      this.buffers.current = new Buffer(gl, {data, usage});
+      this.buffers.next = new Buffer(gl, {data, usage});
+    }
+
+    this.currentLength = toLength;
+
+    this.attributeInTransition.update({
+      buffer: this.buffers.current,
+      // Hack: Float64Array is required for double-precision attributes
+      // to generate correct shader attributes
+      value: this.attribute.value
+    });
+
+    if (this.isTransitioning()) {
+      this.transitionSettings.onInterrupt();
+    }
+
+    this.transform = this.transform || new Transform(gl, getShaders(this.attribute.size));
+    this.transform.update({
+      elementCount: Math.floor(this.currentLength / this.attribute.size),
+      sourceBuffers: {
+        aTo: getSourceBufferAttribute(gl, this.attribute)
+      }
+    });
+  }
+
+  update() {
+    if (!this.isTransitioning()) {
+      return false;
+    }
+
+    // TODO: fire an onStart() event here if the transition has just started
+
+    this.transform.update({
+      sourceBuffers: {
+        aPrev: this.buffers.previous,
+        aCur: this.buffers.current
+      },
+      feedbackBuffers: {
+        vNext: this.buffers.next
+      }
+    });
+    this.transform.run({
+      uniforms: {
+        stiffness: this.transitionSettings.stiffness,
+        damping: this.transitionSettings.damping
+      }
+    });
+    this.buffers = cycleBuffers(this.buffers);
+    this.attributeInTransition.update({buffer: this.buffers.current});
+
+    this.transitionSettings.onUpdate();
+
+    // TODO: fire an onEnd() event here if the transition has just ended
+
+    return true;
+  }
+
+  cancel() {
+    this.transitionSettings.onInterrupt();
+    this.transform.delete();
+    if (this.buffers.previous) this.buffers.previous.delete();
```

Comment:
Nit: just use a for loop to delete & reset to null

Reply:
Would you prefer `buffers` be an array rather than an object with `previous`, `current`, and `next` properties?

Reply:
I went ahead and made the buffers for both transition types an array.